python/pyarrow/tests/test_convert_pandas.py

```diff
@@ -814,6 +811,43 @@ def test_python_datetime(self):
         })
         tm.assert_frame_equal(expected_df, result)
 
+    @pytest.mark.xfail(
+        LooseVersion(pd.__version__) >= "0.23.0",
```

Comment:
For `pandas==0.22` the test works fine. I opened an issue https://github.com/pandas-dev/pandas/issues/21142 but I'm not sure how arrow wants to handle this cc @wesm

Reply:
This seems like a pure Pandas issue to me. We should adopt our expectation of the result to be independent of the underlying Pandas problem by adding `expected_df["datetime"] = pd.to_datetime(expected_df["datetime"])` before the failing assertion.